prime-angular-frontend/src/app/config/config.service.ts

```diff
@@ -160,8 +160,8 @@ export class ConfigService implements IConfigService {
    * @description
    * Sort the configuration by weight.
    */
-  private sortConfigByWeight(): (a: LicenseConfig, b: LicenseConfig) => SortWeight {
-    return (a: LicenseConfig, b: LicenseConfig) =>
-      this.utilsService.sortByKey<LicenseConfig>(a, b, 'weight');
+  private sortConfigByWeight<T extends IWeightedConfig>(): (a: T, b: T) => SortWeight {
+    return (a: T, b: T) =>
+      this.utilsService.sortByKey<T>(a, b, 'weight');
```

Comment:
Feels like we keep adding private methods like `sortConfigByXXX()`  in `ConfigService `class., and it's merely just a shorthand of calling the `sortConfigByKey `method. 

Since these sorting methods are just private methods and not consumed outside of `ConfigService`, it's unlikely to cause problems elsewhere. Why not directly use `sortConfigByKey<T>('name')`, `sortConfigByKey<T>('code')`, and `sortConfigByKey<T>('weight') `? Also, it's strange to have sorting methods in a `ConfigService `class, maybe move somewhere else to be extension methods if really needed.

Reply:
Yah, we can definitely remove those methods out of the config.

Reply:
I just wanted to fix the typings since they weren't generic

Reply:
> I just wanted to fix the typings since I noticed they weren't generic, but can definitely remove them. Do you mind taking that on @johnnyl6?

@mtpultz  I did some refactoring. Let me know if they are okay.

Reply:
Look great.  Thanks @johnnyl6